src/app/api/collections/quality_matrix.py

```diff
@@ -249,25 +249,38 @@ def replication_source_quality_matrix(collection: Tree) -> QualityMatrix:
     #           "metadatacontributer_creator" : { "doc_count" : 55 }
     #         },
     #         ...
+
     row_headers = _replication_source_row_headers()
 
+    Bucket = dict[str, Any]
+
+    def rows() -> Iterator[tuple[QualityMatrixHeader, Optional[Bucket]]]:
+        """Returns None for buckets that are not part of the elastic query result."""
+        # first loop over all replication sources and insert empty rows in case we did not find anything
+        # See: https://github.com/openeduhub/metaqs-main/issues/121
+        buckets = {bucket["key"]: bucket for bucket in response.aggregations["replication_source"]["buckets"]}
+        for key, header in row_headers.items():
+            # remove bucket from dictionary to check what is left after this loop
+            yield header, buckets.pop(key, None)
+        # now, check if there are any buckets left (in case the set of
+        # replication sources from edusharing is not complete)
+        for key, bucket in buckets.items():
+            # use defaults for the rows as we have no alternative...
+            yield QualityMatrixHeader(id=bucket["key"], label=bucket["key"], alt_label=bucket["key"], level=0), bucket
+
     return QualityMatrix(
         rows=[
             QualityMatrixRow(
-                meta=row_headers.get(
-                    bucket["key"],
-                    # default to the plain id as label in case we cannot look it up.
-                    QualityMatrixHeader(id=bucket["key"], label=bucket["key"], alt_label=bucket["key"], level=0),
-                ),
+                meta=header,
                 counts={
                     # return the number of materials where the meta data field is __NOT__ missing.
-                    name: bucket["doc_count"] - bucket[attribute.path]["doc_count"]
+                    name: 0 if bucket is None else (bucket["doc_count"] - bucket[attribute.path]["doc_count"])
```

Comment:
AFAIK understand, we will always get all replication sources which are in the aggregations. I.e., if they have no element in the collection, we see only dashes. Furthermore, if the bucket is empty, we still get an entry. I like

Reply:
If there is a replication source `foo` and there are no materials having that replication source (e.g. because of the additional collection filter), then there will not be a bucket for that replication source.
Hence to always show all replication source rows, we extend the elastic search result with empty rows, where no bucket is present. We know that a bucket is missing, because we have the set of replication sources from the edusharing metadataset endpoint.

If you mean the dashes in the frontend: they are purely frontend side: The backend simply returns total(row) count=0 and cell count=0. to avoid NaN (0/0) the frontend displays a dash.

Reply:
Alright, thanks for the explanation. :)